### DISCUSSION TITLE
404 tracking – plausible is undefined

### DISCUSSION THREAD

**Discussion by NiklasBr** [plausible/analytics#2657] [Anything Else]
We are getting the following error when [trying to implement 404 error tracking](https://plausible.io/docs/error-pages-tracking-404#3-create-a-custom-event-goal-in-your-plausible-analytics-account):
```console
Uncaught ReferenceError: plausible is not defined
    <anonymous> https://localhost/404:712
```

this is the footer on the 404 error template:
```html
	</footer>

	<script>
		window.dataLayer = window.dataLayer || [];
		function gtag(){dataLayer.push(arguments);}
		gtag('consent', 'default', { 'ad_storage': 'denied', });
		gtag('consent', 'default', { 'analytics_storage': 'denied', });
		gtag('consent', 'default', { 'social_media': 'denied', });
		dataLayer.push({ 'event': 'default_consent' });
	</script>
	<script>
		(function(w,d,s,l,i){w[l]=w[l]||[];w[l].push({'gtm.start': new Date().getTime(),event:'gtm.js'});var f=d.getElementsByTagName(s)[0],
			j=d.createElement(s),dl=l!='dataLayer'?'&l='+l:'';j.async=true;j.src='https://www.googletagmanager.com/gtm.js?id='+i+dl;f.parentNode.insertBefore(j,f);
		})(window,document,'script','dataLayer','GTM-REDACTED');
	</script>

	<script src="[https://www.google.com/recaptcha/enterprise.js?hl=nb](view-source:https://www.google.com/recaptcha/enterprise.js?hl=nb)" async defer></script>
	<script>plausible("404", { props: { path: document.location.pathname } });</script>
</body>
</html>
```

The plausible tag is served as as GTM tag:
![Screenshot 2023-02-08 at 11 58 32](https://user-images.githubusercontent.com/279826/217510947-233b0fd7-dd33-40f3-a171-d084afe67d35.png)

What is wrong here?

**Comment by metmarkosaric** [comment]
hi @NiklasBr! our script automatically disables itself when running on localhost as most people don't want those stats to be counted. we do have a manual extension to track localhost here: https://plausible.io/docs/script-extensions#all-our-script-extensions

**Reply by NiklasBr** [reply]
The localhost thing is just for development, this is later supposed to be deployed to a regular website.

I got it working a minute ago by wrapping it in a timeout, is that supposed to work?

```html
    <script>
        setTimeout(function() {
            plausible("404", { props: { path: document.location.pathname } });
        }, 600);
    </script>
```

![Screenshot 2023-02-08 at 12 32 48](https://user-images.githubusercontent.com/279826/217518340-469a7a50-fb91-4772-9294-3ff957f68d5a.png)

Server response was `202 Accepted`

**Reply by metmarkosaric** [reply]
ah that's great to hear, thanks for the update! it should work without it as described in our docs but now i'm not sure if running it through GTM might change things. have you tried without GTM to see if it works without the timeout?

**Reply by NiklasBr** [reply]
Loading scripts directly without GTM and setTimeout() works as expected.

**Reply by metmarkosaric** [reply]
thanks for checking and good to hear that! we've had other examples of GTM adding some complexity which is why we have added that "warning" section to our GTM instructions. sometimes it just breaks things in unexpected ways. but i'm glad that you figured it out with GTM too! we'll keep your tip in mind if someone else has the same issue in the future
